util/helm/client.go

```diff
@@ -355,8 +352,8 @@ func normalizeChartName(chart string) string {
 	return nc
 }
 
-func (c *nativeHelmChart) getCachedChartPath(chart string, version string) string {
-	return path.Join(c.repoPath, fmt.Sprintf("%s-%s.tgz", strings.ReplaceAll(chart, "/", "_"), version))
+func (c *nativeHelmChart) getCachedChartPath(chart string, version string) (string, error) {
+	return c.chartCachePaths.GetPath(fmt.Sprintf("%s/%s-%s", c.repoURL, strings.ReplaceAll(chart, "/", "_"), version))
```

Comment:
Can we just use a struct as the key instead of a string? That would make sure we don't have collisions. 

```go
type RepoKey struct {
    RepoURL string
    Chart string
    Version string
}
```

Reply:
I don't want to keep `TempPaths` generic. Using `RepoKey` would make it too repo-server specific. I wish golang support generics but without it have to use string.

I agree about possible conflicts. As workaround using `:` as URL and chart separator. URL path cannot have `:` so it ensure there will be no conflicts.

Reply:
Looks like we can't rely on a colon: http://en.wikipedia.org/wiki/Template:Welcome

What if, in repo-server, we use `RepoKey` but JSON-encode it before using it as a key? That way the JSON encoding does the delimiting for us.

Reply:
Great idea - implemented json encoded key